### PR TITLE
Switch `httpbin` tests over to JuliaLang-hosted `httpbin` mock server

### PR DESCRIPTION
**Pull request by staticfloat**
This should fix the recent testing issues we've had with `httpbin`.

### CODE REVIEW

File: test/download.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-let urls = ["http://httpbin.org/ip", "https://httpbin.org/ip"]
+let urls = ["http://httpbin.julialang.org/ip", "https://httpbin.julialang.org/ip"]
```
Comment (Keno) [inline test/download.jl:3]: Looks like we only get `https` here.

Comment (Keno) [reply test/download.jl:3]:
```suggestion
let urls = ["http://julialang.org", "https://httpbin.julialang.org/ip"]
```

Comment (Keno) [reply test/download.jl:3]: Also needs to be adjusted below though.

Comment (staticfloat) [reply test/download.jl:3]: I'm not sure what you're saying here; this chunk is only testing that `Base.download_url()` is affected by `Base.DOWNLOAD_HOOKS`; nothing is actually downloaded.

Comment (Keno) [reply test/download.jl:3]: ah, my bad.

Comment (staticfloat) [reply test/download.jl:4]: I've added a comment to make this more explicit.

**Review by Keno — APPROVED** [review]
Should add the API definition in contrib or something, in case we get hit by two badly timed buses and somebody else needs to re-deploy this, but LGTM otherwise.

**Pull request merged**